### PR TITLE
Reorganize span detail components

### PR DESCRIPTION
**Pull request by tiffon**
For now, target branch is `refactor-trace-detail-selectors`, PR #53 is a WIP.

- Split sub-components of `SpanDetail` out into separate files.
- Move styles into CSS (addresses [this PR comment](https://github.com/uber/jaeger-ui/pull/53#discussion_r134318660))
- Misc style and DOM tweaks  (see attached screenshot)
  - Expanded table styling
  - Tag summary styling
  - Background on hover of expandable / collapsable elements
  - Debug info

span-detail-pr-marked

### CODE REVIEW

**Comment by jcorbin** [comment]
Awesome, love the screenshot btw, makes reviewing this at a high/user level much easier.

My only feedback is: what if instead of hiding (how even do we show?) the span id, why not float it in the upper right (across from the Big Bold operation name)?  I'd even be fine if you wanted to soften the grey contrast a bit so it doesn't draw as much attention.

**Comment by yurishkuro** [comment]
I suggest keeping "SpanID:" label before the hex value in bottom right.

Otherwise looks great.

**Comment by tiffon** [comment]
@jcorbin 

> My only feedback is: what if instead of hiding (how even do we show?) the span id, why not float it in the upper right (across from the Big Bold operation name)?

Currently, the span ID is always showing in the lower right corner of the detail panel (the down arrow points to it). Do you mean moving it to the upper right instead of the lower right, or were you thinking it wasn't being shown at all?

**Comment by tiffon** [comment]
Screenshot of "SpanID:" label added.
screen shot 2017-08-22 at 2 43 10 pm

**Comment by jcorbin** [comment]
Ahh okay then, I was only suggesting that upper right seems easier to get to, rather than scroll past one or more expanded-or-collapsed sections. It's also a top-level property, similar to tags or operation name in my mind. Maybe needing span ids isn't a common enough thing to worry about tho... not sure.

**Comment by yurishkuro** [comment]
> Maybe needing span ids isn't a common enough thing to worry about tho..

it isn't, imo, only used by "experts", e.g. when you want to deep dive into trace JSON

File: src/components/App/App.css [hunk 1]
```diff
@@ -24,6 +24,21 @@ THE SOFTWARE.
   height: 100%;
 }
 
+.u-no-float {
+  float: none;
+}
+
+.u-cursor-pointer {
```
Comment (saminzadeh) [inline src/components/App/App.css]: what does the `u` prefix mean?

File: src/components/TracePage/TraceTimelineViewer/SpanDetail/AccordianKeyValues.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import KeyValuesTable from './KeyValuesTable';
+import toggleEnhancer from './toggle-enhancer';
+
+import './AccordianKeyValues.css';
+
+const dataPropType = PropTypes.arrayOf(
```
Comment (saminzadeh) [inline src/components/TracePage/TraceTimelineViewer/SpanDetail/AccordianKeyValues.js]: Flowtype?

File: src/components/TracePage/TraceTimelineViewer/SpanDetail/AccordianLogs.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  logs: PropTypes.arrayOf(
+    PropTypes.shape({
+      timestamp: PropTypes.number,
+      fields: PropTypes.array,
+    })
+  ).isRequired,
+  timestamp: PropTypes.number.isRequired,
+};
+
+export default toggleEnhancer(AccordianLogs);
```
Comment (saminzadeh) [inline src/components/TracePage/TraceTimelineViewer/SpanDetail/AccordianLogs.js]: Good use of HOC here

File: src/components/App/App.css (hunk 1 above)
Comment (tiffon) [inline src/components/App/App.css]: @saminzadeh util

File: src/components/TracePage/TraceTimelineViewer/SpanDetail/AccordianLogs.js (hunk 3 above)
Comment (tiffon) [inline src/components/TracePage/TraceTimelineViewer/SpanDetail/AccordianLogs.js]: @saminzadeh Thanks

File: src/components/App/App.css (hunk 1 above)
Comment (saminzadeh) [inline src/components/App/App.css]: Got it, I recommend bringing this in for CSS utils:
http://basscss.com/

File: src/components/TracePage/TraceTimelineViewer/SpanDetail/AccordianKeyValues.js (hunk 2 above)
Comment (tiffon) [inline src/components/TracePage/TraceTimelineViewer/SpanDetail/AccordianKeyValues.js]: Adding flow typing, which is handy for props, definitely reminded me why I'm skeptical — bugs.

- [Changes](https://github.com/uber/jaeger-ui/pull/64/commits/5c7f896ed4b39836af91b81995bd0b18f0cf0f60#diff-cbab2020bb1863553129838da2e79e95) that gave me trouble
- [Underlying issue](https://github.com/facebook/flow/issues/1511)
- [Simplified example](https://goo.gl/4Z23MJ)

**Pull request merged**